### DISCUSSION TITLE
Flashing Zigbee Bridge Pro (ESP32 based) with Tasmota

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#18515] [Q&A] [answered]
Hello,
I am planning on flashing a new Sonoff ZigBee BridgePRO with Tasmota. I have few question: According to "NotEnoughTech" website ESP32 Flash tool is to be used to flash the device.  

(1) Can I use "Tasmotizer 1.2" to do the flash instead? 

(2) If yes, then in "Tasmotizer" when selecting Flash size of backup for "Save original Firmware" what value should I select? 1MB or 4MB? I am unable to find/figure out original firmware size.

Edit: I am planning on using  Unofficial Tasmota32-ZigbeeBridge.factory.bin from [https://github.com/tasmota/install/blob/firmware/firmware/unofficial/tasmota32-zigbeebridge.factory.bin](url)

Thanks in Advance.

**Comment by sfromis** [comment, marked as answer]
No, Tasmotizer does not work for ESP32. Use the [Tasmota Web installer](https://tasmota.github.io/install/). And do not use that "ESP32 Flash tool" either.

The article by "NotEnoughTech" is how it was before an enhanced build made it **much** easier, as installing the correct image includes a pre-populated file system with the needed stuff ready to use. You should use the image `tasmota32-zbbrdgpro.factory.bin` which is what you get when selecting "Tasmota32 Sonoff-ZigbeeBridgePro" in the Tasmota Web installer.

Roughly, the text from NotEnoughTech becomes useful starting at "Flashing coordinator firmware on Sonoff ZB Bridge Pro"

To back up original firmware first, I'd just use standard `esptool.py`. Flash chip size is 4MB.

**Reply by NsinghP** [reply]
Hardest part was figuring out how to backup using esptool.py, but figure out I did! Installing python and using esptool.py from command line in windows11 was a challenge. But then one line gave me the original firmware backup I was looking for: "esptool.py -b 115200 --port COM3 write_flash --flash_freq 80m 0x000000 flash_4M.bin" <---I have saved the file and I don't think I'm going to need it.  Rest of your instructions worked perfectly

![Screenshot (206)](https://user-images.githubusercontent.com/69470381/234766627-39bc5cd9-635e-4a52-93c4-f6a0aeeedc13.png)


Now on to configuring Tasmota on Sonoff ZigBee Bridge Pro.
